packages/collector/Collector.php

```diff
@@ -0,0 +1,250 @@
+
+add_action('admin_menu', 'collector_plugin_menu');
+add_action('plugins_loaded', 'collector_plugins_loaded');
+add_filter('plugin_install_action_links', 'collector_plugin_install_action_links', 10, 2);
+add_filter('plugins_api_args', 'collector_plugins_api_args', 10, 2);
+
+function get_collector_admin_page_url() {
+	return admin_url(COLLECTOR_DOWNLOAD_PATH);
+}
+
+function collector_plugins_loaded()
+{
+	if(!current_user_can(ADMIN_PAGE_CAPABILITY))
+	{
+		return;
+	}
+
+	if(home_url($_SERVER['REQUEST_URI']) === get_collector_admin_page_url())
+	{
+		collector_zip_collect();
+		exit();
+	}
+}
+
+function collector_plugin_menu()
+{
+	add_submenu_page(
+		NULL,
+		__('Collector', TRANSLATE_DOMAIN),
+		__('Collector', TRANSLATE_DOMAIN),
+		ADMIN_PAGE_CAPABILITY,
+		COLLECTOR_ADMIN_PAGE_SLUG,
+		'collector_render_playground_page',
+		NULL
+	);
+}
+
+function collector_render_playground_page()
+{?>
+	<div id="wp-playground-wrapper">
+		<div id="wp-playground-toolbar">
+			<span>
+				<?php
+				printf(
+					__(
+						'WordPress Playground preview for %s',
+						TRANSLATE_DOMAIN
+					),
+					get_bloginfo('name')
+				);
+				?>
+			</span>
+			<a href="<?php echo admin_url( 'plugin-install.php' ); ?>" id = "goBack">
+				<?php _e('Go Back', TRANSLATE_DOMAIN); ?>
+			</a>
+		</div>
+		<div id="wp-playground-main-area">
+			<iframe id="wp-playground"></iframe>
+		</div>
+	</div>
+	<script type = "text/javascript">
+		const frame  = document.getElementById('wp-playground');
+		const zipUrl = '<?php echo esc_url( get_collector_admin_page_url() ); ?>';
+
+		const query = new URLSearchParams(window.location.search);
+
+		const username   = '<?php echo esc_js(wp_get_current_user()->user_login); ?>';
+		const fakepass   = '<?php echo esc_js(collector_get_fakepass()); ?>';
+		const blueprintUrl = query.get('blueprintUrl');
+		(async () => {
+			const  { startPlaygroundWeb } = await import('<?php echo esc_url(COLLECTOR_PLAYGROUND_PACKAGE);?>');
+			const blueprint = await (await fetch(blueprintUrl)).json();
+
+			blueprint.steps = blueprint.steps || [];
+			blueprint.steps = [
+				...blueprint.steps,
+				{
+					step: 'writeFile',
+					path: '/data.zip',
+					data: {
+						'resource': 'url',
+						'url': zipUrl,
+					},
+				},
+				{
+					step: 'unzip',
+					zipPath: '/data.zip',
+					extractToPath: '/wordpress',
+				},
+				{
+					step: 'rm',
+					path: '/data.zip',
+				},
+				{
+					step: 'runSql',
+					sql: {
+						resource: 'vfs',
+						path: '/wordpress/schema/_Schema.sql',
+					}
+				},
+				{
+					step: 'login',
+					username: username,
+					password: fakepass,
+				}
+			];
+
+			blueprint.preferredVersions = {
+				wp: '<?php echo esc_js(COLLECTOR_WP_VERSION); ?>',
+				php: '<?php echo esc_js(COLLECTOR_PHP_VERSION); ?>',
+			};
+
+			const client = await startPlaygroundWeb({
+				iframe: document.getElementById('wp-playground'),
+				remoteUrl: `https://playground.wordpress.net/remote.html`,
```

Comment:
This is fine for V1, but eventually let's put base URL in a config file and enable local overrides for working with a local Playground dev server.

Reply:
I added filters to both Playground urls in d4cb48d53d1646b345fa62002bdd548854f4911b